lib/test/lis/types.cpp

```diff
@@ -130,49 +131,312 @@ TEST_CASE("16-bit floating point", "[type]") {
-        153,
-        -153,
-    };
+    SECTION("standard floats") {
+        const std::array< bytes<4>, 5 > inputs = {{
+            { 0x00, 0x00, 0x00, 0x00 }, // 0
+            { 0x00, 0x08, 0x4C, 0x80 }, // 153
+            { 0x00, 0x08, 0xB3, 0x80 }, // -153
+            { 0xFF, 0xFF, 0x40, 0x00 }, // 0.25
+            { 0xFF, 0xFF, 0xC0, 0x00 }, // -0.25
+        }};
+
+        const std::array< float, inputs.size() > expected = {
+            0,
+            153,
+            -153,
+            0.25,
+            -0.25,
+        };
+
+        for( std::size_t i = 0; i < expected.size(); ++i ) {
+            float v;
+            lis_f32low( inputs[ i ], &v );
+            CHECK( v == expected[ i ] );
+        }
+    }
+
+    /*
+     * Low-resolution float numbers do not fit under float (or even double).
+     * Tests below attempt to define the boundaries of that discrepancy.
+     *
+     * It's difficult to assure that presented numbers are actually
+     * the max/min/smallest architecture-represented ones, not last due to
+     * differences in space assigned to exponent and fraction, but they
+     * should be somewhere near.
+     */
+
+    SECTION(" near-maximum architecture-represented positive number ") {
+        /* E = 2^7, M = 2^15 - 1 */
+
+        /*
+         * Expected is stored explicitly because some compilers overflow float
+         * when calculating.
+         *
+         * (1 - std::pow(2.0f, -15)) * std::pow(2.0f, std::pow(2.0f, 7))
+         */
+        const float expected = 340271982327221393808117546439109771264.0f;
+        const bytes<4> input = { 0x00, 0x80, 0x7F, 0xFF };
 
-    for( std::size_t i = 0; i < expected.size(); ++i ) {
         float v;
-        lis_f32low( inputs[ i ], &v );
-        CHECK( v == expected[ i ] );
+        lis_f32low( input, &v );
+        CHECK( v == expected );
+        CHECK (v <= std::numeric_limits<float>::max());
+    }
+
+    SECTION(" near-minimum architecture-represented negative number ") {
+        /* E = 2^7, M = -(2^15 - 1) */
+
+        /*
+         * Expected is stored explicitly because some compilers overflow float
+         * when calculating.
+         *
+         * -(1 - std::pow(2.0f, -15)) * std::pow(2.0f, std::pow(2.0f, 7))
+         */
+        const float expected = -340271982327221393808117546439109771264.0f;
+        const bytes<4> input = { 0x00, 0x80, 0x80, 0x01 };
+
+        float v;
+        lis_f32low( input, &v );
+        CHECK( v == expected );
+        CHECK (v >= std::numeric_limits<float>::lowest());
+    }
+
+    SECTION(" maximum LIS-stored positive number (data loss)") {
+        /* E = 2^15 - 1, M = 2^15 - 1 */
+        const float expected =
+            (1 - std::pow(2.0f, -15)) * std::pow(2.0f, std::pow(2.0f, 15) - 1);
+        const bytes<4> input = { 0x7F, 0xFF, 0x7F, 0xFF };
+
+        float v;
+        lis_f32low( input, &v );
+        CHECK( v == expected );
+        // Failing: too big number, can't be represented by float
```

Comment:
Why are these here?

Reply:
Boundaries definition.
It's important to know the range of the data we do and do not support to have a clear picture.

I would have uncommented and marked the section as xfail if catch2 supported this option of marking sections as failed. Opted out for commented checks instead.